### PR TITLE
Make https the default for all mirrors

### PR DESCRIPTION
**Pull request by xshadow**
It think it would be a good idea to make https the default for all mirrors.

What do you think?

### CODE REVIEW

**Comment by tersmitten** [comment]
It's probably a good idea, but I don't think it works out of the box. You need to have HTTPS transport installed and that's not always the case.

**Comment by xshadow** [comment]
You are absolutely right, in older Debian versions there is the need for apt-transport-https .


But if you think it is in general a good idea, I could work on that and also corresponding molecule testing.

File: defaults/main.yml [hunk 1]
```diff
@@ -7,20 +7,21 @@ apt_backports_enable: true
 
 # Ubuntu specific
 apt_ubuntu_mirror: mirror://mirrors.ubuntu.com/mirrors.txt
-apt_ubuntu_security_mirror: http://security.ubuntu.com/ubuntu
```
Comment (tersmitten) [inline defaults/main.yml:10]: Unfortunately this still does not work (for Ubuntu)

File: templates/etc/apt/sources.list.Ubuntu.j2 [hunk 2]
```diff
@@ -56,13 +56,13 @@ deb {{ apt_ubuntu_security_mirror }} {{ ansible_distribution_release }}-security
 # # This software is not part of Ubuntu, but is offered by Canonical and the
 # # respective vendors as a service to Ubuntu users.
 {% if apt_ubuntu_partner_enable %}
-deb http://archive.canonical.com/ubuntu {{ ansible_distribution_release }} partner
```
Comment (tersmitten) [inline templates/etc/apt/sources.list.Ubuntu.j2:59]: Unfortunately this still does not work (for Ubuntu)

File: templates/etc/apt/sources.list.Ubuntu.j2 [hunk 3]
```diff
@@ -56,13 +56,13 @@ deb {{ apt_ubuntu_security_mirror }} {{ ansible_distribution_release }}-security
...
-{{ '# ' if not apt_src_enable else '' }}deb-src http://archive.canonical.com/ubuntu {{ ansible_distribution_release }} partner
+deb https://archive.canonical.com/ubuntu {{ ansible_distribution_release }} partner
+{{ '# ' if not apt_src_enable else '' }}deb-src https://archive.canonical.com/ubuntu {{ ansible_distribution_release }} partner
 {% endif %}
 
 # # This software is not part of Ubuntu, but is offered by third-party
 # # developers who want to ship their latest software.
 {% if apt_ubuntu_extras_enable and ansible_distribution_version is version('16.04', '<') %}
-deb http://extras.ubuntu.com/ubuntu {{ ansible_distribution_release }} main
```
Comment (tersmitten) [inline templates/etc/apt/sources.list.Ubuntu.j2:66]: Unfortunately this still does not work (for Ubuntu)

**Comment by tersmitten** [comment]
See #50

**Pull request closed without merging**